Flaky integration tests: timing-sensitive patterns cause CI failures

## Problem

Integration tests intermittently fail in CI due to timing-sensitive patterns. The most recent failure was `TestScenario_NearHomeDepartureDoesNotOpenGarage` in PR #931 ([run 23725611469](https://github.com/NickBorgers/home-automation/actions/runs/23725611469)):

```
--- FAIL: TestScenario_NearHomeDepartureDoesNotOpenGarage (0.74s)
    Error: Expected nil, but got: &testutil.ServiceCall{Domain:"cover", Service:"open_cover",
           ServiceData:map[string]interface{}{"entity_id":"cover.garage_door_door"}}
```

## Known Flaky Tests

### 1. `TestScenario_NearHomeDepartureDoesNotOpenGarage` (scenario_security_test.go:351)

**Root cause:** Race between departure timestamp recording and near_home suppression check. When `nick_home` → off fires, the statetracking plugin records `nickDepartureTime`. 1 minute later, `nick_near_home` → on fires, but under CI load the departure timestamp handler may not have completed before the near_home handler reads it, causing the cooldown check to miss and the garage to open.

**Introduced by:** PR #918 (departure cooldown feature, commit d5aba6f)

### 2. Tests using `time.Sleep()` instead of deterministic waits

| Test | File | Sleep |
|------|------|-------|
| `TestScenario_RapidSensorChanges...` | scenario_concurrent_triggers_test.go:212 | 20ms |
| `TestScenario_WakeupZoneResolution...` | scenario_wakeup_zone_resolution_test.go:161-164 | 30ms |

Under CI load, 20-30ms may be insufficient for state propagation. Should use `waitForBoolState()` or `waitForProcessing()`.

### 3. Documented race windows requiring manual state sequencing

- scenario_wakeup_zone_resolution_test.go:522 — comment documents a race: `anyone_asleep` must be set and propagated BEFORE `master_asleep` to avoid a timing window
- scenario_multi_plugin_test.go:267-268 — relies on `waitForProcessing()` for handler ordering

## Suggested Fixes

1. **Add `waitForProcessing()` after `server.SetState("input_boolean.nick_home", "off", nil)` in TestScenario_NearHomeDepartureDoesNotOpenGarage** — ensures departure timestamp is recorded before near_home fires
2. **Replace `time.Sleep(20ms)` / `time.Sleep(30ms)` with `waitForProcessing()` or `waitForCondition()`** in concurrent trigger and zone resolution tests
3. **Consider whether `t.Parallel()` is safe** for tests using mock clocks — each test gets its own instance but goroutine scheduling under parallel load affects handler execution order

## Files

- `homeautomation-go/test/integration/scenario_security_test.go`
- `homeautomation-go/test/integration/scenario_concurrent_triggers_test.go`
- `homeautomation-go/test/integration/scenario_wakeup_zone_resolution_test.go`
- `homeautomation-go/internal/plugins/statetracking/manager.go` (departure tracking at lines ~385-387, ~505-507)

--- a/homeautomation-go/test/integration/scenario_concurrent_triggers_test.go
+++ b/homeautomation-go/test/integration/scenario_concurrent_triggers_test.go
@@ -3,7 +3,6 @@ package integration
 import (
 	"context"
 	"testing"
-	"time"
 
 	"homeautomation/internal/plugins/lighting"
 	"homeautomation/internal/plugins/music"
@@ -204,12 +203,13 @@ func TestScenario_RapidPresenceChanges_StableState(t *testing.T) {
 
 	// Simulate rapid presence changes
 	for i := 0; i < 5; i++ {
+		expectedHome := i%2 == 0
 		if i%2 == 0 {
 			env.server.SetState("input_boolean.caroline_home", "on", map[string]interface{}{})
 		} else {
 			env.server.SetState("input_boolean.caroline_home", "off", map[string]interface{}{})
 		}
-		time.Sleep(20 * time.Millisecond) // Intentional: simulates rapid sensor changes
+		waitForBoolState(t, env.stateManager, "isCarolineHome", expectedHome, "presence change %d should propagate", i)
 	}
 
 	// Final state: Caroline is home (last toggle was i=4, even → on)

--- a/homeautomation-go/test/integration/scenario_security_test.go
+++ b/homeautomation-go/test/integration/scenario_security_test.go
@@ -368,6 +368,7 @@ func TestScenario_NearHomeDepartureDoesNotOpenGarage(t *testing.T) {
 	t.Log("WHEN: Nick leaves home (home zone clears first)")
 	server.SetState("input_boolean.nick_home", "off", nil)
 	waitForBoolState(t, manager, "isNickHome", false, "isNickHome should be false after departure")
+	waitForProcessing(t, manager)
 
 	snapshot := server.ServiceCallCount()
 

--- a/homeautomation-go/test/integration/scenario_wakeup_zone_resolution_test.go
+++ b/homeautomation-go/test/integration/scenario_wakeup_zone_resolution_test.go
@@ -158,9 +158,9 @@ func TestScenario_WakeUp_DebouncesRapidTriggers(t *testing.T) {
 	// These three changes would each independently trigger zone resolution
 	// without debouncing. With debouncing, they should coalesce into one.
 	env.server.SetState("input_boolean.anyone_asleep", "off", map[string]interface{}{})
-	time.Sleep(30 * time.Millisecond) // Intentional: simulates real timing between state changes
+	waitForBoolState(t, env.stateManager, "isAnyoneAsleep", false, "isAnyoneAsleep should update before next wake event")
 	env.server.SetState("input_boolean.master_asleep", "off", map[string]interface{}{})
-	time.Sleep(30 * time.Millisecond) // Intentional: simulates real timing between state changes
+	waitForBoolState(t, env.stateManager, "isMasterAsleep", false, "isMasterAsleep should update before next wake event")
 	env.server.SetState("input_boolean.wake_sequence_active", "on", map[string]interface{}{})
 
 	// Wait for debounce timer to fire using channel synchronization

